vertx-context/src/test/java/io/smallrye/common/vertx/ContextLocalsTest.java

```diff
@@ -0,0 +1,120 @@
+package io.smallrye.common.vertx;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.assertj.core.api.Assertions;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import io.vertx.core.Context;
+import io.vertx.core.Vertx;
+import io.vertx.core.impl.ContextInternal;
+
+public class ContextLocalsTest {
+
+    private static Vertx vertx;
+
+    @BeforeClass
+    public static void initVertx() {
+        vertx = Vertx.vertx();
+    }
+
+    @AfterClass
+    public static void closeVertx() {
+        if (vertx != null) {
+            vertx.close().toCompletionStage().toCompletableFuture().join();
+        }
+    }
+
+    @Test
+    public void assertAccessFromRootContext() throws InterruptedException {
+        CountDownLatch latch = new CountDownLatch(3);
+
+        vertx.runOnContext(x -> {
+            Assertions.assertThatThrownBy(() -> {
+                ContextLocals.get("foo");
+            }).isInstanceOf(UnsupportedOperationException.class);
+
+            latch.countDown();
+        });
+
+        vertx.runOnContext(x -> {
+            Assertions.assertThatThrownBy(() -> {
+                ContextLocals.put("foo", "bar");
+            }).isInstanceOf(UnsupportedOperationException.class);
+
+            latch.countDown();
+        });
+
+        vertx.runOnContext(x -> {
+            Assertions.assertThatThrownBy(() -> {
+                ContextLocals.remove("foo");
+            }).isInstanceOf(UnsupportedOperationException.class);
+
+            latch.countDown();
+        });
+
+        Assertions.assertThat(latch.await(100, TimeUnit.MILLISECONDS)).isTrue();
```

Comment:
Nice way to propagate assertion status from the event loop thread to the main thread :-)

Reply:
I'd be +1 to move to JUnit 5, obviously.

Reply:
I've switched to Vert.x Junit 5.